### ISSUE TITLE
Serializing [Enum:String]

### ISSUE THREAD

**Issue by hegra** [evermeer/EVReflection#106]
Serializing enum based on strings (EVRawString, EVAssociated ) and serializing arrays of strings work fine. But serializing arrays of enums based on strings only result in value "true".

**Comment by evermeer** [comment]
Just to be clear, is a class like this where the serializing goes wrong?

```
class WorkaroundObject: EVObject, EVArrayConvertable {   
    enum StatusType: Int, EVRawInt {
        case NotOK = 0
        case OK = 1
    }

    var enumType: StatusType = .OK
    var enumList: [StatusType] = []

    override func setValue(value: AnyObject!, forUndefinedKey key: String) {
        switch key {
        case "enumType":
            if let rawValue = value as? Int {
                if let status =  StatusType(rawValue: rawValue) {
                    self.enumType = status
                }
            }
        default:
            self.addStatusMessage(.IncorrectKey, message: "SetValue for key '\(key)' should be handled.")
            print("---> setValue for key '\(key)' should be handled.")
        }
    }
}
```

And it will go wrong for the property enumList?

Indeed you will get a log message like:
ERROR: valueForAny unkown type Array<StatusType> for value: [....
StatusType does not have an NSObject as it's base class and therefore cannot easily be created.
I have to investigate if I can find a workaround for that.

**Comment by hegra** [comment]
Yes but as type String not Int

**Comment by evermeer** [comment]
For using an array of enums a fix plus a workaround was needed.
Mainly because Swift does not allow you to create a NSArray that contains a swift enum and it's not possible to create an instance of an Enum just based on a string description of it's type. 
I have just pushed a fix plus a workaround unit test as version 2.40.0 to GitHub and Cocoapods.

You can find the unit test that will show you how it can be used here:
https://github.com/evermeer/EVReflection/blob/030adbdd89ee9a12ca252ba26f967eba02759adb/EVReflection/EVReflectionTests/EVReflectionWorkaroundsTests.swift#L86-L86
This test is for an Array of enums that have an Int as it's rawValue. If the rawValue is a String, then it works the same.

When you want to keep the enum array, then you need the following:
- Your object needs to implement the EVArrayConvertable for the convertArray function
- You need to implement the 'setValue forUndefinedKey' for the enum array

**Comment by hegra** [comment]
Works great with strings based enums also
